### PR TITLE
Add process launch APIs to dbgshim

### PR DESCRIPTION
**Pull request by mikem8361**
Added CreateProcessForLaunch, ResumeProcess and CloseResumeHandle that can be better supported cross platform.

### CODE REVIEW

**Comment by mikem8361** [comment]
@noahfalk  and @jkotas could you review?

File: src/dlls/dbgshim/dbgshim.cpp [hunk 1]
```diff
@@ -62,6 +62,90 @@ if it exists, it will:
     } \
     CONTRACTL_END; \
 
+
+//-----------------------------------------------------------------------------
+// Public API.
+//
+// CreateProcessForLaunch - a stripped down version of the Windows CreateProcess
+// that can be supported cross-platform.
+//
```
Comment (noahfalk) [inline src/dlls/dbgshim/dbgshim.cpp]: What sort of error handling is expected? I'm guessing right now its just success = TRUE, failure = FALSE and no further info is provided?

File: src/dlls/dbgshim/dbgshim.cpp [hunk 2]
```diff
@@ -62,6 +62,90 @@ if it exists, it will:
...
+        lpCurrentDirectory,
+        &startupInfo,
+        &processInfo);
+
+    if (processInfo.hProcess != NULL)
+    {
+        CloseHandle(processInfo.hProcess);
+    }
+
+    *pProcessId = processInfo.dwProcessId;
```
Comment (noahfalk) [inline src/dlls/dbgshim/dbgshim.cpp]: Should we trust these processInfo fields if result == FALSE? Perhaps we should just zero the output values instead to be certain.

File: src/dlls/dbgshim/dbgshim.cpp [hunk 3]
```diff
@@ -62,6 +62,90 @@ if it exists, it will:
...
+    *pResumeHandle = processInfo.hThread;
+
+    return result;
+}
+
+//-----------------------------------------------------------------------------
+// Public API.
+//
+// ResumeProcess - to be used with the CreateProcessForLaunch resume handle
+//
```
Comment (noahfalk) [inline src/dlls/dbgshim/dbgshim.cpp]: We should document what the return value is

File: src/dlls/dbgshim/dbgshim.cpp [hunk 4]
```diff
@@ -62,6 +62,90 @@ if it exists, it will:
...
+    return ResumeThread(hResumeHandle);
+}
+
+//-----------------------------------------------------------------------------
+// Public API.
+//
+// CloseResumeHandle - to be used with the CreateProcessForLaunch resume handle
+//
+//-----------------------------------------------------------------------------
+DWORD
```
Comment (noahfalk) [inline src/dlls/dbgshim/dbgshim.cpp]: document return value

**Comment by mikem8361** [comment]
@noahfalk to address your feedback, I changed the return type for the new APIs to HRESULT to be consistent with the rest of the dbgshim API.

**Comment by mikem8361** [comment]
@dotnet-bot Ubuntu x64 Checked Build and Test

**Pull request merged**